src/store/ChartStore.js

```diff
@@ -728,6 +728,26 @@ class ChartStore {
         this.stxx.chart.seriesRenderers._main_series.seriesParams[0].field = 'Close';
     }
 
+    destroyFeed() {
+        if (this.feed && !Object.keys(this.feed._isSubscriptionInitializing).length) {
+            this.feed.unsubscribeAll();
+            this.feed = null;
+            if (ChartStore.keystrokeHub.context === this.context) {
+                ChartStore.keystrokeHub.setActiveContext(null);
+            }
+            if (this.stxx) {
+                this.stxx.container.removeEventListener('mouseenter', this.onMouseEnter);
+                this.stxx.container.removeEventListener('mouseleave', this.onMouseLeave);
+                this.stxx.updateChartData = function () {}; // prevent any data from entering the chart
+                this.stxx.isDestroyed = true;
+                this.stxx.destroy();
+                this.stxx = null;
```

Comment:
@nazaninreihani It'd be a good practice to remove all appends and prepends functions to `CIQ.ChartEngine` here too.
```suggestion
                CIQ.ChartEngine.prototype.remove('headsUpHR');
                CIQ.ChartEngine.prototype.remove('resizeChart');
                this.stxx = null;
```